src/main/java/Parser.java

```diff
@@ -0,0 +1,160 @@
+    static final String TASKLIST_SAVE_PATH = "data/saveTaskList.txt";
+
+    public static void printLine() {
+        System.out.print(LINEBREAK);
+    }
+
+    static List list = new List();
+
+    public static void Instance() {
+        int saveNumber = 0;
+
+        //checking if there is a save file and accessing it
+        File savedList = new File(TASKLIST_SAVE_PATH);
+        boolean doesSaveExist = savedList.exists();
+        try {
+            if (!doesSaveExist) {
+                savedList.getParentFile().mkdirs();
+                savedList.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("☹ OOPS!!! I am unable to create a new directory or file :-(");
+        }
+        try {
+            Scanner saveFileScanner = new Scanner(savedList);
+
+            while (saveFileScanner.hasNext()) {
+                String currentRead = saveFileScanner.nextLine();
+                String[] taskSave = currentRead.trim().split(" \\| ");
+                switch (taskSave[0].trim()) {
+                    case "todo":
+                        saveNumber++;
+                        list.addTaskFromSave(taskSave[2]);
+                        if (taskSave[1].equals("1")) {
+                            list.markDoneFromSave(saveNumber);
+                        }
+                        break;
+                    case "deadline":
+                        saveNumber++;
+                        list.addDeadlineFromSave(taskSave[2], taskSave[3]);
+                        if (taskSave[1].equals("1")) {
+                            list.markDoneFromSave(saveNumber);
+                        }
+                        break;
+                    case "event":
+                        saveNumber++;
+                        list.addEventFromSave(taskSave[2], taskSave[3]);
+                        if (taskSave[1].equals("1")) {
+                            list.markDoneFromSave(saveNumber);
+                        }
+                        break;
+                }
+            }
+
+        } catch (FileNotFoundException e) {
+            System.out.println("☹ OOPS!!! I am unable to read the file :-(");
+        }
+
+        //main while loop that takes in the user input
+        String input = scanner.nextLine().trim();
+
+        while (!input.equals(EXITCOMMAND)) {
+            String[] stringTokens = input.split(" ", 2);
+            String command = stringTokens[0];
+            boolean isValidCommand = false;
+            printLine();
+            try {
+                isValidCommand = isCommand(command);
+            } catch (UnrecognizedCommandException e) {
+                System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+            if (isValidCommand) {
+                switch (command) {
+                    case "list":
+                        list.printList();
+                        break;
+                    case "done":
```

Comment:
same goes here... maybe you should improve your code abstraction by having a single method (in charge for the abstraction) for each case statement